tests/api_tests.py

```diff
@@ -151,6 +155,11 @@ def test_infrared_sensor(self):
         self.assertEqual(val,               [])
 
     def test_medium_motor_write(self):
+
+        # brickpi/brickpi3 only support LargeMotor
+        if ev3dev2.get_current_platform() in ("brickpi", "brickpi3"):
```

Comment:
Is the issue here that running the tests on-device doesn't use the `fake` platform? I'd prefer to fix that rather than working around the detected platform.

Reply:
It uses the fake platform, the problem is on a brickpi/brickpi3 we raise now raise an Exception if any Motor class other than LargeMotor, is used.

Reply:
But if it's using the fake platform how does it know it's on a BrickPi?

Reply:
I see what you are saying now...let me think about this for a bit

Reply:
My recollection was that the solution you implemented for the test was not to use the `fake` platform, but instead to manually import some things from `fake` in the tests. I think we should override the detected platform in tests.

Reply:
We could have ``get_current_platform()`` check for a ``USE_FAKE_SYS`` env variable that is set by api_tests.py.

Reply:
Sounds like that would work.